### DISCUSSION TITLE
Does this library has any component to display code snippet?

### DISCUSSION THREAD

**Discussion by fingers10** [microsoft/fluentui-blazor#3043] [Q&A]
For example, like highlight.js - https://highlightjs.org/?

**Comment by vnbaaij** [comment]
We have a `CodeSnippet` component in the demo site (which uses highlight.js). Source for the demo is here in the repo in the examples folder.
It is not part of the library itself, but you can copy the implementation, of course.

**Comment by fingers10** [comment]
@vnbaaij  I was checking the implementation.. I'm confused on line no https://github.com/microsoft/fluentui-blazor/blob/dev/examples/Demo/Shared/Components/DemoSection.razor.cs#L123 .

Can you help me understand how this works?

**Reply by vnbaaij** [reply]
We create `.txt` files of all our `.razor` and `.cs` exmple files. During build we copy those to a specific folder and we created the StaticAssetsService to read those files that works both for Blazor Server as for Blazor WASM.

If you clone the repo locally, you should be able to run FluentUI.Demo Client or Server project locally and debug/step through
 it
